actions/orchestration.go

```diff
@@ -108,17 +108,64 @@ func (o *rdsOrchestrator) databaseModify(c buffalo.Context, id string, input *Da
 
 	if input.Cluster != nil {
 		input.Cluster.DBClusterIdentifier = aws.String(id)
+
+		// set default cluster parameter group when upgrading engine version
+		if input.Cluster.DBClusterParameterGroupName == nil && input.Cluster.EngineVersion != nil {
+			// get information about the existing cluster to determine the engine type
+			describeClusterOutput, err := o.client.Service.DescribeDBClustersWithContext(c, &rds.DescribeDBClustersInput{
+				DBClusterIdentifier: aws.String(id),
+			})
+			if err == nil && describeClusterOutput != nil {
+				pgFamily, pgErr := o.client.DetermineParameterGroupFamily(describeClusterOutput.DBClusters[0].Engine, input.Cluster.EngineVersion)
+				if pgErr != nil {
+					log.Println(pgErr.Error())
+					return nil, pgErr
+				}
+				log.Println("Determined ParameterGroupFamily based on Engine:", pgFamily)
+				cPg, ok := o.client.DefaultDBClusterParameterGroupName[pgFamily]
+				if !ok {
+					log.Println("No matching DefaultDBClusterParameterGroupName found in config, using AWS default PG")
+				} else {
+					log.Println("Using DefaultDBClusterParameterGroupName:", cPg)
+					input.Cluster.DBClusterParameterGroupName = aws.String(cPg)
+				}
+			}
+		}
```

Comment:
maybe these should be a separate function where you pass *input kinda like the ecs-api does?